### PR TITLE
Rebuild MS.CA.VisualBasic

### PR DESCRIPTION
**Pull request by jaredpar**
Fixed two issues:
- The way VB preporocessor symbols are encoded and decoded
- Syntax files were using the on disk path which can have casing
  differences from the PDB path, flipped to PDB path

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL

File: src/Tools/BuildValidator/BuildValidator.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
+    <TargetFrameworks>net472;netcoreapp3.1</TargetFrameworks>
```
Comment (jaredpar) [inline src/Tools/BuildValidator/BuildValidator.csproj:7]: This is done to match the ordering of `TargetFrameworks` in the csc / vbc project files. This way when F5 between them and BuildValidator locally you're using the same runtime. Important for inner dev loop.

File: src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb [hunk 2]
```diff
@@ -767,13 +767,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
-                                                                                              If (p.Value Is Nothing) Then
-                                                                                                  Return p.Key
-                                                                                              End If
-
-                                                                                              Return p.Key + "=" + p.Value.ToString()
-                                                                                          End Function)
+                Dim preprocessorStrings = Options.ParseOptions.PreprocessorSymbols.Select(
+                    Function(p) As String
+                        If TypeOf p.Value Is String Then
+                            Return p.Key + "=""" + p.Value.ToString() + """"
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb:773]: >.ToString() [](start = 58, length = 11)

Is `.ToString()` necessary?

**Review by cston — APPROVED** [review]

File: src/Tools/BuildValidator/BuildConstructor.cs [hunk 3]
```diff
@@ -264,15 +282,20 @@ private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions
...
-            var preprocessorSymbols = string.IsNullOrEmpty(define)
-                ? Array.Empty<KeyValuePair<string, object>>()
-                : define.Split(',')
-                    .Select(s => s.Split('='))
-                    .Select(a => new KeyValuePair<string, object>(a[0], a[1]))
-                    .ToArray();
+            IReadOnlyDictionary<string, object>? preprocessorSymbols = null;
+            if (OptionToString(CompilationOptionNames.Define) is string defineString)
+            {
+                preprocessorSymbols = VisualBasicCommandLineParser.ParseConditionalCompilationSymbols(defineString, out var diagnostics);
```
Comment (RikkiGibson) [inline src/Tools/BuildValidator/BuildConstructor.cs:288]: should we just allow this diagnostic bag to propagate out, rather than passing one in?

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Looks like `VisualBasicDeterministicBuildCompilationTests` baselines need to be updated

File: src/Tools/BuildValidator/BuildConstructor.cs (hunk 3 above)
Comment (jaredpar) [reply src/Tools/BuildValidator/BuildConstructor.cs:288]: I started with than and changed the return to `(Compilation? Compilation, DiagnosticBag? DiagnosticBag)` but I felt the consumption code was very awkward to write. Eventually I went with the approach of passing a bag down as we do in other parts of the code and it felt a bit more natural.

**Pull request merged**